src/include/migraphx/op/pointwise.hpp

```diff
@@ -45,23 +45,18 @@ struct pointwise
         {
             MIGRAPHX_THROW("should have one submodule.");
         }
-        auto* pm = mods.front();
-        if(pm->get_output_shapes().size() != 1)
-            MIGRAPHX_THROW("pointwise should have only one output.");
         if(inputs.empty())
             MIGRAPHX_THROW("pointwise should have at least one input");
+        auto* pm    = mods.front();
         auto pnames = pm->get_parameter_names();
-        std::sort(pnames.begin(), pnames.end());
         check_shapes{inputs, *this}.has(pnames.size()).same_dims();
 
-        auto type = pm->get_output_shapes().front().type();
-
-        // Scalar output if all inputs are scalar
-        if(inputs.front().elements() == 1 and
-           all_of(inputs, [](const auto& s) { return s.scalar(); }))
-            return shape{type};
-
-        return shape::from_permutation(type, inputs.front().lens(), find_permutation(inputs));
```

Comment:
Each pointwise operator should propagate the transpose. This will ensure the fused pointwise will have the same layout as the non-fused pointwise.

Reply:
> . This will ensure the fused pointwise will have the same layout as the non-fused pointwise. 

This may result in different output layout compared to previous logic.  I am wondering about performance impact. 

e.g. pointwise module doing add between two transposed inputs. 
previous logic would return output shape that is transposed. 
Current logic from what i understand would instead return standard shape.

Reply:
>  Current logic from what i understand would instead return standard shape.

No it wont return standard shape. The `binary` operator will return the permutation of the non-packed or non-broadcasted tensor:

https://github.com/ROCm/AMDMIGraphX/blob/bd52f7172abb76722283fcb9660990cf94db7190/src/include/migraphx/op/binary.hpp#L81